File: kotlin/src/main/kotlin/io/github/devstarsj/kotlinbackend/ItemDto.kt
package io.github.devstarsj.kotlinbackend

import java.time.LocalDateTime

class ItemDto {
    val id: Long?
    val name: String
    val price: Int
    val userId: Long?
    val createdAt: LocalDateTime
    val updatedAt: LocalDateTime

    constructor(item: Item) {
        id = item.id
        name = item.name
        price = item.price
        userId = item.userId
        createdAt = item.createdAt
        updatedAt = item.updatedAt
    }
}

File: nest.js/src/dtos/itemDto.ts
import { ObjectType, Field } from '@nestjs/graphql'
import { UserDto } from './userDto'

@ObjectType()
export class ItemDto {

  @Field()
  readonly id?: number

  @Field()
  name: string

  @Field()
  price: number

  // @Field()
  // userId?: number

  @Field(type => UserDto, { nullable: true })
  user?: UserDto

  @Field()
  createdAt: Date

  @Field()
  updatedAt: Date
}
